operators/src/adapters/raster_subquery_adapter.rs

```diff
@@ -619,9 +630,9 @@ where
                     .tile_geo_transform()
                     .grid_idx_to_upper_left_coordinate_2d(i)
             })
+            .filter(|c| dbg!(valid_bounds.contains_coordinate(c)))
```

Comment:
```suggestion
            .filter(|c| valid_bounds.contains_coordinate(c))
```

Maybe also add a line why this is done. I guess it has to do with the set of points that determine the other qrect.

Reply:
as we have a raster here which is aligned to the axes, can't we more efficiently filter the coordinates by intersecting the whole raster tile with the area of use and only check the pixels in the intersection?